SG14/slot_map.h

```diff
@@ -114,11 +114,7 @@ class slot_map
     // O(1) time and space complexity.
     //
     constexpr reference at(const key_type& key) {
-        auto value_iter = this->find(key);
-        if (value_iter == this->end()) {
-            SLOT_MAP_THROW_EXCEPTION(std::out_of_range, "at");
-        }
-        return *value_iter;
+        return const_cast<reference>(static_cast<const slot_map*>(this)->at(key));
```

Comment:
The built-in `const_cast` is not guaranteed to be able to cast between `const_reference` and `reference`, unless they happen to be native references — which might be essentially required anyway (I think it is, for allocator types, at least?) but I'd rather avoid `const_cast` anyway.

Reply:
The `const_cast` is required for this to work. It is one of the rare cases where `const_cast` is safe. Are you thinking of an alternative?

Reply:
huh, I had no idea `vector<bool>` references behaved that way. I don't think a solution to that particular problem exists. I'll close the PR then as there is nothing else to salvage.